font-weight should be clamped to 1 as minimum

darinadler opened the pull request:
#### 00db99db9acfe127589e95787d0b8923146676a9
<pre>
font-weight should be clamped to 1 as minimum
<a href="https://bugs.webkit.org/show_bug.cgi?id=246575">https://bugs.webkit.org/show_bug.cgi?id=246575</a>
rdar://problem/101441591

Reviewed by Tim Nguyen.

* LayoutTests/fast/text/font-weight-parse-expected.txt: Updated to expect PASS.
* LayoutTests/fast/text/font-weight-parse.html: Changed to expect a minimum of 1 rather than of 0.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-shorthand-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-weight-parsing-expected.txt:
Updated to expect more PASS.

* LayoutTests/platform/gtk/fast/text/font-weight-parse-expected.txt: Removed.
* LayoutTests/platform/ios/fast/text/font-weight-parse-expected.txt: Removed.
* LayoutTests/platform/mac/fast/text/font-weight-parse-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/text/font-weight-parse-expected.txt: Removed.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeFontWeightNumberRaw): Clamp to [1,1000].

Canonical link: <a href="https://commits.webkit.org/255873@main">https://commits.webkit.org/255873@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/50207949fa9e88a5bd8124cb61248d6b8e5fd79b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93865 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3057 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24440 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103497 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/163831 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97857 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3074 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31292 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86185 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99502 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99532 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2195 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80289 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29216 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84121 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72168 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37688 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17651 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35552 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18914 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4055 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39431 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41483 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38144 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
```diff
@@ -1548,8 +1548,6 @@ RefPtr<CSSPrimitiveValue> consumeNumberOrPercent(CSSParserTokenRange& range, Val
-
 #if !ENABLE(VARIATION_FONTS)
     auto isIntegerAndDivisibleBy100 = [](double value) {
         ASSERT(value > 0 && value <= 1000);
```
I think this can be changed to `>= 1` according to the spec: https://w3c.github.io/csswg-drafts/css-fonts/#font-weight-absolute-values

darinadler replied:
OK, you convinced me. I spotted this obvious mistake, but I didn’t want to change untested code, but I guess I’ll do it. We really need to delete this unused code path soon. I wonder what’s holding us back at this point.

nt1m commented on Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
```diff
@@ -1568,7 +1566,7 @@ std::optional<double> consumeFontWeightNumberRaw(CSSParserTokenRange& range)
         if (!(result->value > 0 && result->value < 1000) || !isIntegerAndDivisibleBy100(result->value))
```
ditto here

nt1m reviewed (approved):
I guess my two comments don't matter too much since it's for `!ENABLE(VARIATION_FONTS)`, but it's probably harmless and an improvement to do the change anyway.

darinadler committed:
font-weight should be clamped to 1 as minimum
https://bugs.webkit.org/show_bug.cgi?id=246575
rdar://problem/101441591

Reviewed by Tim Nguyen.

* LayoutTests/fast/text/font-weight-parse-expected.txt: Updated to expect PASS.
* LayoutTests/fast/text/font-weight-parse.html: Changed to expect a minimum of 1 rather than of 0.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-shorthand-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/variations/font-weight-parsing-expected.txt:
Updated to expect more PASS.

* LayoutTests/platform/gtk/fast/text/font-weight-parse-expected.txt: Removed.
* LayoutTests/platform/ios/fast/text/font-weight-parse-expected.txt: Removed.
* LayoutTests/platform/mac/fast/text/font-weight-parse-expected.txt: Removed.
* LayoutTests/platform/wpe/fast/text/font-weight-parse-expected.txt: Removed.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeFontWeightNumberRaw): Clamp to [1,1000].

Canonical link: https://commits.webkit.org/255873@main
